lib/spack/spack/spec.py

```diff
@@ -3384,7 +3384,8 @@ def __init__(self, spec, hash):
 class NoSuchHashError(SpecError):
     def __init__(self, hash):
         super(NoSuchHashError, self).__init__(
-            "No installed spec matches the hash: '%s'")
+            "No installed spec matches the hash: '%s'"
+        % hash)
```

Comment:
It works as is.  Are parentheses preferred when they seem-to-be/are optional?

Thought about changing it to the format form because I've seen you prefer it in other places, but then I'd feel obliged to convert the rest of the file and that way lies madness and new bugs....